Extracted testing helpers to standalone file to use them in integration tests too

Fed03 opened the pull request:
I've extracted the `calendarCenter` and `calendarSelect` test helpers to their own files and put them in `addon-test-support` folder.

Now we can swith to explicit import and use them like this
```js
import { calendarSelect, calendarCenter } from 'ember-power-calendar/test-support';
```
This allows to use them also in integration tests with the only prerequisite being
```js
import { initCalendarHelpers } from 'ember-power-calendar/test-support';

moduleForComponent('component', {
  integration: true,
  beforeEach() {
    initCalendarHelpers(this.container);
  }
});
```
For acceptance tests, this manual setup is unecessary caouse it's handled by an initializer

The change is not breaking, I've added deprecation notices though.

cibernox commented on tests/acceptance/helpers-test.js:
```diff
@@ -1,76 +1,185 @@
 });
 
-test('`calendarCenter` invokes the `onCenterChange` action of the target component WITH the selector we pass it', async function(assert) {
+test('`calendarCenter` invokes the `onCenterChange` action of the target component WITH the selector we pass it', async function(
```
Why this indentation chenge?

Fed03 replied:
I think prettier auto formatted these following eslint rules. If you want I'll try to revert it

cibernox replied:
I'd prefer so.

Fed03 replied:
Done

Fed03 commented:
Any news on this? as always there's no hurry :)

cibernox commented:
I've been moving countries so I had zero-time in the last few weeks. I hope that will change next week.

Fed03 commented:
I've fixed an issue with the initializer. It referenced a piece of code placed in `test-support` therefore in non testing envs it errored.

I removed the initializer, the only downside is the manual initialization step in acceptance test aswell
```js
import { initCalendarHelpers } from 'ember-power-calendar/test-support';

moduleForAcceptance('foo', {
  beforeEach() {
    initCalendarHelpers(this.application);
  }
});
```

I still think that injecting the application is unnecessary, If you can look into it maybe we could remove the initialization step altogether.

cibernox commented on test-support/helpers/ember-power-calendar.js:
```diff
@@ -1,59 +1,25 @@
+      false,
+      { id: 'ember-power-calendar.test-helpers.calendarSelect', until: '1.0.0' }
+    );
+    initCalendarHelpers(app.__container__);
```
Do you know if we can use `Ember.getOwner(app)` here?

Fed03 replied:
I already tried but seems that app got no owner

cibernox commented on test-support/helpers/ember-power-calendar.js:
```diff
@@ -1,59 +1,25 @@
-  let calendarService = app.__container__.lookup('service:power-calendar');
-  return calendarService._calendars[calendarGuid];
-}
+import { deprecate } from '@ember/application/deprecations';
```
Today the right thing seems to be `import { deprecate } from '@ember/debug';`

cibernox commented on addon-test-support/calendar-center.js:
```diff
@@ -0,0 +1,26 @@
+    !!onCenterChange
+  );
+  let publicAPI = calendarComponent.get('publicAPI');
+  run(async () => {
```
I *believe* you can do `await run(() => publicAPI.actions.changeCenter(newCenter, publicAPI));`

cibernox commented on addon-test-support/calendar-center.js:
```diff
@@ -0,0 +1,26 @@
+import Ember from 'ember';
+import { assert } from 'ember-metal/utils';
```
`import { assert } from '@ember/debug';`

cibernox commented on addon-test-support/calendar-center.js:
```diff
@@ -0,0 +1,26 @@
+import Ember from 'ember';
```
`import run from '@ember/runloop';`

cibernox commented on addon-test-support/calendar-select.js:
```diff
@@ -0,0 +1,24 @@
+import moment from 'moment';
+import { assert } from 'ember-metal/utils';
```
`import { assert } from '@ember/debug';`

cibernox commented on addon-test-support/calendar-test-service.js:
```diff
@@ -0,0 +1,54 @@
+import { assert } from 'ember-metal/utils';
```
`import { assert } from '@ember/debug';`

Fed03 commented:
I'll create a dummy component to add the integration tests you requested asap

wynnfarm commented:
Any chance this branch will be merged soon? It would be nice to leverage this for integrated component testing.

cibernox commented:
Sorry, I forgot about this completely. I barely remember what this was about. I'll try to review this tomorrow. If I haven't answered in 48h, ping me again.

chuckcarpenter commented:
ping @cibernox ?

cibernox commented:
@Fed03 @chuckcarpenter I actually found a better way. I started developing it in https://github.com/cibernox/ember-power-calendar/compare/make_helpers_usable_in_integration?expand=1 but it will require the new testing approach in http://rwjblue.com/2017/10/23/ember-qunit-simplication/

I should finish it today.

Fed03 commented:
so you've found out how to remove the dependency from application, good!
lemme know if I can be of any help

cibernox commented:
@Fed03 I might have, but now I have to do some actual work. I'll come back to this tonight.

Fed03 commented:
@cibernox no problem mate, real life first of all

cibernox commented:
@Fed03 I think this has be closed by #105 🎉

Fed03 commented:
@cibernox I'll test it as soon as possible :sparkler:

cibernox commented:
Released along with another bugfix in 0.6.0
